src/System.Runtime/tests/System/ActivatorTests.cs

```diff
@@ -64,41 +66,202 @@ public static void CreateInstance_ConstructorWithParamsParameter()
+            Assert.NotNull(Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: true));
+
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor)));
+            Assert.Throws<MissingMethodException>(() => Activator.CreateInstance(typeof(ClassWithPrivateDefaultConstructor), nonPublic: false));
+        }
+
+        [Fact]
+        public void CreateInstance_NullableType_ReturnsNull()
+        {
+            Assert.Null(Activator.CreateInstance(typeof(int?)));
+        }
+
+        [Fact]
+        public void CreateInstance_NullType_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("type", () => Activator.CreateInstance((Type)null));
+            AssertExtensions.Throws<ArgumentNullException>("type", () => Activator.CreateInstance(null, new object[0]));
+        }
+
+        [Fact]
+        public static void CreateInstance_MultipleMatchingConstructors_ThrowsAmbiguousMatchException()
+        {
             Assert.Throws<AmbiguousMatchException>(() => Activator.CreateInstance(typeof(Choice1), new object[] { null }));
+        }
 
+#if netcoreapp || uapaot
+        [Fact]
+        public void CreateInstance_NotRuntimeType_ThrowsArgumentException()
+        {
+            foreach (Type nonRuntimeType in Helpers.NonRuntimeTypes)
+            {
+                AssertExtensions.Throws<ArgumentException>("type", () => Activator.CreateInstance(nonRuntimeType));
+                AssertExtensions.Throws<ArgumentException>("type", () => Activator.CreateInstance(nonRuntimeType, new object[0]));
+            }
+        }
+#endif // netcoreapp || uapaot
+
+        public static IEnumerable<object[]> CreateInstance_ContainsGenericParameters_TestData()
+        {
+            yield return new object[] { typeof(List<>) };
+            yield return new object[] { typeof(List<>).GetTypeInfo().GenericTypeParameters[0] };
+        }
+        
+        [Theory]
+        [MemberData(nameof(CreateInstance_ContainsGenericParameters_TestData))]
+        public void CreateInstance_ContainsGenericParameters_ThrowsArgumentException(Type type)
+        {
+            AssertExtensions.Throws<ArgumentException>(null, () => Activator.CreateInstance(type));
+            AssertExtensions.Throws<ArgumentException>(null, () => Activator.CreateInstance(type, new object[0]));
+        }
+
+        public static IEnumerable<object[]> CreateInstance_InvalidType_TestData()
+        {
+            yield return new object[] { typeof(void) };
+            yield return new object[] { typeof(void).MakeArrayType() };
+            yield return new object[] { Type.GetType("System.ArgIterator") };
+            // Fails with TypeLoadException in .NET Core
+            // yield return new object[] { Type.GetType("System.ArgIterator").MakeArrayType() };
```

Comment:
We should check this in just commented out.  If there's an issue for it, we should include the issue # here.  If it's by design, we should delete these lines.